internal/scripts/ci_build_and_run_end_to_end_tests.sh

```diff
@@ -83,7 +84,7 @@ if [[ ${BUILD_END_TO_END_TESTS} == true ]]; then
         -DCMAKE_BUILD_TYPE=Debug \
         -DBUILD_CXX=ON \
         -DBUILD_EXAMPLES=ON \
-        -DBUILD_TESTING=OFF
+        -DBUILD_TESTING=ON
```

Comment:
Okay, now I remember again why I added the separate cmake flag. I wanted to prevent to have to build the C++ unit tests just to run the end-to-end-tests. There are other CI jobs which take much longer to run and the C++ unit tests should build quite fast. Semantically, this is the better approach, though. Let's keep it.